week04/139. Word Break.cpp

```diff
@@ -0,0 +1,29 @@
+class Solution {
+public:
+    //TC:O(n^3)n is the length of input string
+    //SC:O(n)
+    bool wordBreak(string s, vector<string>& wordDict) {
+        set<string> word_set(wordDict.begin(), wordDict.end());
+        queue<int> q;
+        vector<bool> visited(s.length(), false);
+        q.push(0);
+        while (!q.empty()) {
+            int start = q.front();
+            q.pop();
+            if (visited[start]) {
+                continue;
+            }
+            for (int end = start + 1; end <= s.length(); end++) {
+                if (word_set.find(s.substr(start, end - start)) !=
+                    word_set.end()) {
+                    q.push(end);
+                    if (end == s.length()) {
+                        return true;
+                    }
+                }
+            }
+            visited[start] = true;
+        }
+        return false;
+    }
+};
```

Comment:
This is not close to an optimal solution. Kindly refer to our editorial for a better understanding.